frontend/packages/ux-editor/src/components/TextResource/TextResource.tsx

```diff
@@ -46,7 +46,7 @@ export const TextResource = ({
   const prevFormItemId = usePrevious(formItemId);
   const prevFormLayoutName = usePrevious(formLayoutName);
 
-  const [currentValue, setCurrentValue] = useState<string>('');
+  const [currentValue, setCurrentValue] = useState<string>(useTextResourceValue(textResourceId));
```

Comment:
Using a hook inside the initializer of a useState should be avoided since the initial value are only supposed to be run once. Using a hook will result in the code for the initial value running on every re-render, which is not efficient, and can lead to issues 🫣 
We should do like this instead:
```suggestion
  const initialTextResourceValue = useTextResourceValue(textResourceId)
  const [currentValue, setCurrentValue] = useState<string>(initialTextResourceValue);
```

Reply:
Fixed